docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,1554 @@
+
+In the [OVN-Kubernetes BGP Integration
+enhancement](https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/docs/okeps/okep-5296-bgp.md#future-goals),
+no-overlay mode was briefly discussed. In this enhancement, we aim to describe
+the feature in detail, define the API changes we want to introduce for it, and
+address a number of concerns with respect to the existing BGP Integration and
+User-Defined Network Segmentation features.
+
+Avoiding Geneve encapsulation and using the provider network for east-west
+traffic stems from the need to minimize network overhead and maximize
+throughput. Users who intend to enable BGP on their clusters can leverage
+BGP-learned routes to achieve this. The goal is to provide users with an API to
+create networks (default or CUDN), with no-overlay mode enabled, allowing
+traffic to skip Geneve encapsulation (i.e., the overlay network) and simply make
+use of the learned routes in the underlay or provider network for inter-node
+communication.
+
+While BGP is the proposed solution for exchanging east-west routing information
+within a cluster, it is not the only possible option. Future implementations of
+no-overlay mode may incorporate additional routing technologies and protocols
+beyond BGP.
+
+## User-Stories/Use-Cases
+
+### Story 1: Have the cluster default network in the no-overlay mode
+
+As a cluster admin, I want to enable no-overlay mode for the cluster default
+network to integrate seamlessly with existing BGP-capable networks, achieve
+maximum network performance, and reduce overhead.
+
+### Story 2: Create a CUDN in the no-overlay mode
+
+As a cluster admin, I want to enable no-overlay mode for a cluster user defined
+network to integrate seamlessly with existing BGP-capable networks and achieve
+maximum network performance.
+
+### Story 3: Deploy networks in the no-overlay mode without external BGP routers
+
+As a cluster admin, I want to use no-overlay mode for intra-cluster traffic
+without advertising pod networks to the external network or depending on
+external BGP routers.
+
+## Proposed Solution
+
+This solution leverages the existing BGP feature to advertise each node's pod
+subnet via the CNCF project [FRR-K8s](https://github.com/metallb/frr-k8s). FRR
+routers managed by FRR-k8s then propagate these routes throughout the provider
+network, enabling direct pod-to-pod traffic without an overlay. A single cluster
+can simultaneously have networks in both overlay and no-overlay modes.
+
+When no-overlay mode is enabled for a network:
+
+* By default, when BGP advertisement is enabled, the BGP feature disables SNAT
+  for pod egress traffic leaving nodes. For users who only want to have
+  no-overlay mode but do not want to expose pod IPs externally, this behavior is
+  not ideal. To address this, users will be able to enable SNAT for
+  pod-to-external traffic. This will be configured via a new `outboundSNAT`
+  field in the `ClusterUserDefinedNetwork` CRD for CUDNs, and the
```

Comment:
Just a note for my future self: That CUDN in itself when created is always going to SNAT pod to external to nodeIP.
What introduces the no-SNAT concept is the RA CRD, however the field to override that is being added via CUDN CRD.
Reasoning for this is because we want the granularity for this per network I guess. The network overlay itself definitely belongs to the network API, just the outboundSNAT is slightly misplaced but I am ok with the current proposal.